src/Integration.UnitTests/Exclusions/ExclusionSettingsStorageTests.cs

```diff
@@ -201,6 +201,48 @@ public void SaveSettings_NoError_SavesSettings()
             file.Verify(f => f.WriteAllText(ExpectedExclusionsFilePath, SerializedExclusions));
         }
 
+        [TestMethod]
+        public void SettingsExist_StandAloneMode_ThrowsInvalidOperationException()
+        {
+            var file = new Mock<IFile>();
+            var bindingConfiguration = BindingConfiguration.Standalone;
+
+            var testSubject = CreateTestSubject(file.Object, bindingConfiguration);
+
+            using var scope = new AssertIgnoreScope();
+
+            Action act = () => testSubject.SettingsExist();
+
+            act.Should().Throw<InvalidOperationException>();
+            file.Invocations.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void SettingsExist_ConnectedMode_FileDoesNotExist_False()
+        {
+            var file = new Mock<IFile>();
+            file.Setup(f => f.Exists(ExpectedExclusionsFilePath)).Returns(false);
+
+            var testSubject = CreateTestSubject(file.Object);
+
+            var result = testSubject.SettingsExist();
+
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void SettingsExist_ConnectedMode_FileExists_True()
+        {
+            var file = new Mock<IFile>();
+            file.Setup(f => f.Exists(ExpectedExclusionsFilePath)).Returns(true);
+
+            var testSubject = CreateTestSubject(file.Object);
+
+            var result = testSubject.SettingsExist();
+
+            result.Should().BeTrue();
+        }
+
```

Comment:
I'm debating if it's worth adding 3 additional tests:
1. Failed to access the file (non critical ex)
2. Failed to access (critical)
3. File exists but content inside is invalid

I think the right approach for all 3 would be to delete the existing file from disk (if we can) and return that the settings don't exist. But I don't think that's what we're doing in other cases (i.e. cfamily). Returning that the file doesn't exist when in fact we couldn't access the file feels wrong to me. So I think it's better not to handle anything here and let the caller (UnboundSolutionChecker) handle the exceptions.

Reply:
We haven't had problems using this approach for the CFamily, so far I'd start simple for now.